src/nix/all_deps.rs

```diff
@@ -0,0 +1,75 @@
+use std::fmt;
+
+use anyhow::{bail, Result};
+
+use super::devour_flake::DrvOut;
+
+/// Nix derivation path
+#[derive(Debug, Ord, PartialOrd, Eq, PartialEq, Clone, Hash)]
+pub struct Drv(pub String);
+
+/// Encompasses both derivation and derivation output paths
+#[derive(Debug, Ord, PartialOrd, Eq, PartialEq, Clone, Hash)]
+pub enum DrvPaths {
+    DrvOut(DrvOut),
+    Drv(Drv),
+}
+
+impl fmt::Display for DrvPaths {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            DrvPaths::DrvOut(drv_out) => write!(f, "{}", drv_out.0),
+            DrvPaths::Drv(drv) => write!(f, "{}", drv.0),
+        }
+    }
+}
+
+/// Query the deriver of an output path
+pub async fn nix_store_query_deriver(out_path: DrvOut) -> Result<Drv> {
+    let nix_store = crate::nix_store_cmd().await;
+    let mut cmd = nix_store.command();
+    cmd.args(["--query", "--deriver", &out_path.0]);
+    nix_rs::command::trace_cmd(&cmd);
+    let drv_out = cmd.output().await?;
+    if drv_out.status.success() {
+        let drv = String::from_utf8(drv_out.stdout)?.trim().to_string();
+        Ok(Drv(drv))
+    } else {
+        let exit_code = drv_out.status.code().unwrap_or(1);
+        bail!(
+            "nix-store --query --deriver failed to run (exited: {})",
+            exit_code
+        );
+    }
+}
+
+/// Query the requisites of a derivation path, including outputs
```

Comment:
This comment doesn't say anything new in addition to what the function name already implies. Can it be rewritten so that a Rust developer familiar with Nix but not yet familiar with `nix-store ...` CLI can understand what it does?